library-best-practices/JLBP-12.md

```diff
@@ -1,8 +1,16 @@
-[JLBP-12] Make level of support and level of API stability clear
-----------------------------------------------------------------
+[JLBP-12] Make level of support and API stability clear
+-------------------------------------------------------
 
-- Surface stability should be clearly described.
-- Feedback mechanisms should be outlined (for example, file an issue).
-- Clearly describe when a library is abandoned.
-- Clearly label 0.x libraries as unstable (as in, they can experience
-  breakages and are not recommended for production use).
+- Clearly describe the lifecycle status (and corresponding level of expected support) of the library. Examples include:
+  - Incubating (no support expected)
+  - Stable (SLOs in place to handle issues)
+  - Maintenance only (only bugs will be addressed)
+  - Deprecated (no work is expected, use a replacement instead)
+  - Abandoned (no work is expected, no clear replacement)
```

Comment:
These two don't fully match our internal instructions. See go/gcp-github  and https://github.com/googlearchive/

I also feel like there's one more relevant page I'm having trouble digging up ATM.

Reply:
I'm trying to keep the JLBPs generic enough to be used by non-Google libraries. Ideally, if there is a non-Google library we want to depend on, we should be able to check that library against these best practices. I'm currently thinking we might want to create a separate set of "GJLBPs" (Google Java Library Best Practices) that refer to Google-specific things, such as the requirement to sign binaries with Google's signature. 

I will merge these two into a single item so that it is less specific. The key point of this best practice is that library owners should make it clear to a human that they shouldn't expect additional work; it doesn't have to follow a rigid classification to comply with this best practice. For Google libraries, we might want to be more rigid, but I think that's separate.